### ISSUE TITLE
Cannot debug processAlgorithm

### ISSUE THREAD

**Issue by eggachecat** [lmotta/debug_vs_plugin#11]
I don't know why, but it ignores all the breakpoints in processAlgorithm function while others in other functions are fine. 
Why is this happening? Is this function being executed in a separate process(operation system process, not QGIS process)?

**Comment by eggachecat** [comment]
![image](https://user-images.githubusercontent.com/18111656/229734753-e146c0e5-22a7-4cbf-b4ae-573edd1d8acd.png)

**Comment by eggachecat** [comment]
still do not know why but `breakpoint()` works for me

**Comment by oeilnc** [comment]
yes, I had the same trouble. here are some explanation and adaptation of the code. 
[doc](https://gist.github.com/AsgerPetersen/2000eb6f3e3307bd25190b19493dd9a3#note-on-processing-plugins)

I prefered Invoke a debug adapter in the processing thread (processAlgorithm) with that code 
```
import debugpy
debugpy.breakpoint()
```
then all reds bullet point works
